go/clustermetadata/topo/store.go

```diff
@@ -393,21 +411,16 @@ func (ts *store) ConnForCell(ctx context.Context, cell string) (Conn, error) {
 		return nil, err
 	}
 
-	serverAddrsStr := strings.Join(ci.ServerAddresses, ",")
-
-	// Return a cached client if present and configuration hasn't changed.
-	ts.mu.Lock()
-	defer ts.mu.Unlock()
+	ts.cellMu.Lock()
+	defer ts.cellMu.Unlock()
 	cc, ok := ts.cellConns[cell]
 	if ok {
-		// Client exists in cache. Verify that it's for the same cell configuration.
-		// The cell name can be reused with different ServerAddresses and/or Root,
-		// in which case we should get a new connection and update the cache.
-		cellAddrs := strings.Join(cc.Cell.ServerAddresses, ",")
-		if serverAddrsStr == cellAddrs && ci.Root == cc.Cell.Root {
+		// Verify that the connection parameters match.
+		if proto.Equal(ci, cc.Cell) {
```

Comment:
The change here looks good. But I'm not sure if we have a test for this. Did you verify this works?